wasm/src/project_system.rs

```diff
@@ -355,25 +356,50 @@ impl From<PackageInfo> for qsc_project::PackageInfo {
 }
 
 /// This returns the common parameters that the compiler/interpreter uses
+#[allow(clippy::type_complexity)]
 #[allow(clippy::needless_pass_by_value)]
 pub(crate) fn into_qsc_args(
     program: ProgramConfig,
     entry: Option<String>,
-) -> (
-    qsc::SourceMap,
-    qsc::TargetCapabilityFlags,
-    qsc::LanguageFeatures,
-) {
+) -> Result<
+    (
+        qsc::SourceMap,
+        qsc::TargetCapabilityFlags,
+        qsc::LanguageFeatures,
+        qsc::PackageStore,
+        Vec<(qsc::hir::PackageId, Option<Arc<str>>)>,
+    ),
+    Vec<qsc::compile::Error>,
+> {
     let capabilities = qsc::target::Profile::from_str(&program.profile())
         .unwrap_or_else(|()| panic!("Invalid target : {}", program.profile()))
         .into();
 
     let pkg_graph: PackageGraphSources = program.packageGraphSources().into();
     let pkg_graph: qsc_project::PackageGraphSources = pkg_graph.into();
 
-    let (sources, language_features) = pkg_graph.into_sources_temporary();
+    // This builds all the dependencies
+    let buildable_program = BuildableProgram::new(capabilities, pkg_graph);
```

Comment:
It struck me how usefly this comment is to remind the reader that `BuildableProgram::new` actually does quite a bit of compiling... at the very least it's worth putting a description into the doc comment on `new` so folks can get that reminder on hover. As food for thought for future work: is there another name besides `new` that can convey how heavyweight this is without being too different from idomatic Rust?